toxbat/requirements.py

```diff
@@ -62,13 +74,15 @@ def are_requirements_changed(config):
 
     def build_fpath_for_previous_version(fname):
         tox_dir = config.config.toxworkdir.strpath
-        fname = '{0}.{1}.previous'.format(fname.replace('/', '-'), config.envname)
+        envdirkey = _str_to_sha1hex(str(config.envdir))
+        fname = '{0}.{1}.previous'.format(fname.replace('/', '-'), envdirkey)
         return os.path.join(tox_dir, fname)
 
     requirement_files = map(parse_requirements_fname, deps)
     return any(
         is_changed(reqfile, build_fpath_for_previous_version(reqfile))
-        for reqfile in requirement_files if reqfile and os.path.isfile(reqfile))
+        for reqfile in requirement_files
```

Comment:
That was my logical error in the code, I intentionally used `any` with generator.